### PR TITLE
Let LocalFile.internalDelete return IStatus instead of boolean

### PR DESCRIPTION
**Pull request by fedejeanne**
This PR will help identifying the underlying error that's causing [#275](https://github.com/eclipse-platform/eclipse.platform.ui/issues/275) by replacing the vague `MultiStatus` returned by `LocalFile::delete` with a more describing `IStatus` created inside `internalDelete`.

### CODE REVIEW

File: resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/LocalFile.java [hunk 1]
```diff
@@ -246,36 +270,44 @@ private boolean internalDelete(File target, String pathToDelete, MultiStatus sta
...
+
+				deleteChildrenStatus.add(internalDelete(new java.io.File(childName), childName, monitor));
+
 				monitor.worked(1);
 			}
+
+			// Abort if one of the children couldn't be deleted.
+			if (!deleteChildrenStatus.isOK())
+				return deleteChildrenStatus;
+
```
Comment (HannesWell) [inline resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/LocalFile.java:282]: Lets add some braces here too.
```suggestion
			if (!deleteChildrenStatus.isOK()) {
				return deleteChildrenStatus;
			}
```

File: resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/LocalFile.java [hunk 2]
```diff
@@ -246,36 +270,44 @@ private boolean internalDelete(File target, String pathToDelete, MultiStatus sta
...
+			// Try to delete the root directory
 			try {
-				// Don't try to delete the root if one of the children failed.
-				if (!failedRecursive && Files.deleteIfExists(target.toPath()))
-					return true;
+				if (Files.deleteIfExists(target.toPath()))
+					return Status.OK_STATUS;
```
Comment (HannesWell) [inline resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/LocalFile.java:286]:
```suggestion
				if (Files.deleteIfExists(target.toPath())) {
					return Status.OK_STATUS;
				}
```

**Review by HannesWell — COMMENTED** [review]
Looks good, that's a reasonable enhancement.
Just a few suggestions for minor code improvements.

**Comment by fedejeanne** [comment]
Thank you @HannesWell for the suggestions!
I added your changes and also solved the conflicts.

**Review by HannesWell — APPROVED** [review]
Looks good. Thank you.

**Pull request merged**